addons/web/tests/test_pivot_export.py

```diff
@@ -0,0 +1,51 @@
+import io
+import json
+import re
+from zipfile import ZipFile
+
+from odoo import http
+from odoo.tests.common import HttpCase
+
+
+class TestPivotExport(HttpCase):
+
+    def test_export_xlsx_with_integer_column(self):
+        """ Test the export_xlsx method of the pivot controller with int columns """
+        self.authenticate('admin', 'admin')
+        jdata = {
+            'title': 'Sales Analysis',
+            'model': 'sale.report',
+            'measure_count': 1,
+            'origin_count': 1,
+            'col_group_headers': [
+                [{'title': 500, 'width': 1, 'height': 1}],
+            ],
+            'measure_headers': [],
+            'origin_headers': [],
+            'rows': [
+                {'title': 1, 'indent': 0, 'values': [{'value': 42}]},
+            ],
+        }
+        response = self.url_open(
+            '/web/pivot/export_xlsx',
+            data={
+                'data': json.dumps(jdata),
+                'csrf_token': http.Request.csrf_token(self),
+            },
+        )
+        response.raise_for_status()
+        xml = ZipFile(io.BytesIO(response.content))
+        required_files = {'xl/sharedStrings.xml', 'xl/worksheets/sheet1.xml'}
+        if not required_files.issubset(set(xml.namelist())):
+            self.fail("Required files are missing in the Excel archive.")
```

Comment:
`set1.issubset(set2)` is the same as `set1 <= set2` so you can actually rewrite it as:
```suggestion
        self.assertLessEqual(required_files, set(xml.namelist), "Required files are missing in the Excel archive.")
```